dashboard/app/controllers/dynamic_config_controller.rb

```diff
@@ -65,4 +66,54 @@ def gatekeeper_set
     flash[:notice] = "Updated successfully! Remember your changes take 30 seconds to go into effect, so don't expect to see the changes immediately on this page."
     redirect_to action: :gatekeeper_show, feature: feature
   end
+
+  def dcdo_show
+    authorize! :read, :reports
+    DCDO.refresh
+
+    @dcdo_hsh = DCDO.to_h
+  end
+
+  def dcdo_set
+    authorize! :read, :reports
+    DCDO.refresh
+
+    params.require([:key, :data_type])
+    key = params[:key]
+    raw_value = params[:value]
+    data_type = params[:data_type]
+
+    begin
+      value =
+        case data_type
+        when "Integer"
+          Integer(raw_value)
+        when "Float"
+          Float(raw_value)
+        when "Boolean"
+          raw_value.to_bool
+        when "String"
+          raw_value
+        else
+          new_value = Oj.load(raw_value)
+          if new_value.class.to_s != data_type
+            raise "#{new_value} does not match data type \"#{data_type}\""
+          end
+          new_value
+        end
```

Comment:
Is there a specific reason why we're doing a bunch of manual type checking rather than just using JSON directly?

Reply:
Yes. Currently DCDO stores everything in DynamoDB as a string. When an engineer manually executes `DCDO.set` they can pass in a value with the correct type: `DCDO.set('my_key', true)` and our dynamic configuration setting converts that to a string before persisting it, and converts it back when returning it.

The web user interface needs to ask the user to explicitly state the datatype of the value so that it can persist it in a way that ensures the datatype is preserved when it's retrieved.

It's been a couple of years since @KatieShipley and I implemented this in a hack-a-thon, so I may have forgotten some of the details, but I'm pretty certain that the core constraint is that we convert all values to string before saving to DynamoDB.